DataFormats/CTPPSDigi/interface/CTPPSPixelDataError.h

```diff
@@ -0,0 +1,56 @@
+#ifndef DataFormats_CTPPSPixelDataError_h
+#define DataFormats_CTPPSPixelDataError_h
+
+//---------------------------------------------------------------------------
+//!  \class CTPPSPixelDataError
+//!  \brief Pixel error -- collection of errors 
+//!
+//!  Class storing info about pixel errors
+//!  
+//---------------------------------------------------------------------------
+
+#include "FWCore/Utilities/interface/typedefs.h"
+
+#include <string>
+
+class CTPPSPixelDataError {
+public:
+
+  /// Default constructor
+  CTPPSPixelDataError();
+  /// Constructor for 32-bit error word
+  CTPPSPixelDataError(uint32_t errorWord32, const int errorType, int fedId);
+  /// Constructor with 64-bit error word and type included (header or trailer word)
+  CTPPSPixelDataError(uint64_t errorWord64, const int errorType, int fedId);
+  /// Destructor
+  ~CTPPSPixelDataError();
+  
+/// create an error message based on errorType
+  void setMessage();				        

```

Comment:
why is this method still needed in the public interface? 
Please move to private.

Given that the value is fully defined by the error type, it would be even better to define a static const string array with the message values and return the right value in the ```errorMessage``` call.
This way we will not waste so much space saving strings in memory and on disk.

Reply:
I've made some changes, please check they satisfy your requirements.